### DISCUSSION TITLE
Indexing more urls

### DISCUSSION THREAD

**Discussion by comsomisha** [MarginaliaSearch/MarginaliaSearch#20] [Ideas]
Due to the closure of Gigablast and Peekier, I suggest adding to the search engine a form for submitting URLs for indexing at the user's request.

**Comment by vlofgren** [comment]
This actually exists, but maybe not super intuitive admittedly. 

If you search for "site:www.example.com", as long as the domain is known by the search engine and isn't indexed or blacklisted for spam, there is a button for adding sites to be crawled. The limitation to known domains is to prevent abuse.

**Comment by comsomisha** [comment]
However, with this approach, new indie sites will be indexed faster in Gemini space (if there is a Gemini version of the site) than in your search engine. It turns out that it is better to create non-commercial sites directly in Gemini.
Limiting indexing to known domains is a terrible decision in the spirit of corporation. Verification of URLs should be carried out after user's  requests.

**Reply by vlofgren** [reply]
It's a regrettable compromise. It's important to realize that the public internet is _extremely adversarial_. Any public endpoint will be abused and there's not a lot that can be done to prevent it.  Bot traffic is so aggressive it's barely possible to have any sort of interactivity these days. I'll still manually add sites if anyone requests it. I sadly can't have a public endpoint for it.

**Reply by comsomisha** [reply]
I am more concerned that your search engine does not notice sites that are related with decentralized technologies and P2P.

**Comment by comsomisha** [comment]
There are some creative ideas for the URL submission form:
1. Accept only text files with URLs, not inserted links
2. Use Cloudflare, as it is already used on the search engine site
3. Place the form to send the URL to the Gemini network
A creative approach will help to handle with the problem of the extremely adversarial internet.

**Reply by vlofgren** [reply]
I've created a repository here where websites can be added in bulk:

https://github.com/MarginaliaSearch/submit-site-to-marginalia-search
